bundler/lib/bundler/checksum.rb

```diff
@@ -9,6 +9,29 @@ class Checksum
     private_constant :DEFAULT_BLOCK_SIZE
 
     class << self
+      # This is used by the lockfile parser to enable checksum validation
+      # when checksums are found in the lockfile.
+      # We don't enable if the user has explicitly disabled checksum validation.
+      # TODO: This can be removed when checksums are enabled by default.
+      # The enable_checksum_validation setting may also be removed once this
+      # is opt-out instead of opt-in
+      def enabled?
+        !Bundler.settings[:disable_checksum_validation] &&
+          Bundler.settings[:enable_checksum_validation]
+      end
+
+      def from_gem_package(gem_package, algo = DEFAULT_ALGORITHM)
+        return nil unless enabled?
```

Comment:
this is a behavior change - we should still be validating the checksums from the server on install by default (unless `:disable_checksum_validation` is set)

Reply:
well, I suppose it wasn't lost until we began disabling this feature. Now we need to add a special case for it.